quarkus-picocli/src/test/java/PicocliProdIT.java

```diff
@@ -0,0 +1,30 @@
+import static java.util.concurrent.CompletableFuture.runAsync;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+
+@Disabled("Requires 'prod' profile. Disabled because the test fails when the application is built and run with the default 'dev' profile. Enable by building with '-Dquarkus.profile=prod'.")
```

Comment:
Should we run the test suite manually? This defeats the whole idea of automated tests.

Reply:
I understand that disabling the test is not ideal, and I believe the reason is clearly explained in the code comments. I've tried several configuration properties, as I mentioned in my earlier comments—including those suggested by Michal—but unfortunately, I couldn't get them to work.

I'm open to any solutions you might have, or I can remove the test if you prefer. What are your suggestions?

Reply:
1. only set `quarkus.profile` in `withProperty`
2. put `withProperty("quarkus.arc.remove-unused-beans", "none")` (which is build time property irrelevant for your test)
3. enable the test

and report back if it fixed your issues

Reply:
I kinda wonder if this can still work considering our FW don't know it is a build time property. There is another way to do this: instead of using `application.properties` you can specify the property using a test class profile. mmnt, I'll check if that can work.

Reply:
yeah, so here is the way:

- use `io.quarkus.test.services.QuarkusApplication#properties` instead of `application.properties`

Try just that for both test classes. If this alone isn't enough, you need to force custom build. The best way to do this is to adjust FW, but you can workaround this by specifying app classes, or forced dependency or the build time config property as I suggested before. All of that must work.

Reply:
I tried that approach on Wednesday but I forgot put a class in the @QuarkusApplication. 
Currently with my changes both tests works, thanks @michalvavrik